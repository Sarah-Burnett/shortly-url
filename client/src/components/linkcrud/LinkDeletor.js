import React, { useState } from "react";
import { LinkModal, PrimaryButton } from "../Styles";
import axios from "axios";
import { shortlyUrl } from "../../utilities/urls";
import Error from "../Error";
import CloseButton from "./CloseButton";
import { useLinks } from "../../hooks/useLinks";
import { useProcessLink } from "../../hooks/useProcessLink";
import { useFlash } from '../../hooks/useFlash';

const LinkDeletor = () => {
  const { deleteLinkFromLinks } = useLinks();
  const {
    processLink: {
      link,
      link: { _id, full, short },
    },
    setProcessNull,
  } = useProcessLink();
  const { showFlash } = useFlash();
  const [error, setError] = useState();
  const updateState = () => {
    deleteLinkFromLinks({ link });
    showFlash('Link deleted');
    setProcessNull();
  };
  const deleteLink = (event) => {
    event.preventDefault();
    axios
      .delete(`/api/links/${_id}`)
      .then((res) => updateState())
      .catch((err) => setError("Delete failed, please try again"));
  };

  return (
    <LinkModal onSubmit={(event) => deleteLink(event)}>
      <h3>Delete Shortly Link</h3>
      <p>{full}</p>
      <p>{shortlyUrl + short}</p>
      {error && <Error error={error} />}
      <PrimaryButton id="delete">Delete</PrimaryButton>
      <CloseButton />
    </LinkModal>
  );
};

export default LinkDeletor;
