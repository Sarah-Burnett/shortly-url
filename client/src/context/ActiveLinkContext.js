import React, { useState, createContext } from 'react';

export const ActiveLinkContext = createContext();

export const ActiveLinkProvider = (props) => {
    const [ activeLink, setActiveLink] = useState({
        activeLink: [],
        process: false
    });

    return (
        <ActiveLinkContext.Provider value={[activeLink, setActiveLink]}>
            {props.children}
        </ActiveLinkContext.Provider>
    )
};