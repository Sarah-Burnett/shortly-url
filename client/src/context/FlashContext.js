import React, { useState, createContext } from 'react';

export const FlashContext = createContext();

export const FlashProvider = (props) => {
    const [ flash, setFlash ] = useState({
        flashMsg: "",
        isFlashVisible: false,
        isAutoHideFlash: true,
    });
    return (
        <FlashContext.Provider value={[flash, setFlash]}>
            {props.children}
        </FlashContext.Provider>
    )
};