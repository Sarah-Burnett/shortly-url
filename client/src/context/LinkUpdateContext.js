import React, { useState, createContext } from 'react';

export const LinkUpdateContext = createContext();

export const LinkUpdateProvider = (props) => {
    const [ linkUpdate, setLinkUpdate] = useState();
    return (
        <LinkUpdateContext.Provider value={[linkUpdate, setLinkUpdate]}>
            {props.children}
        </LinkUpdateContext.Provider>
    )
};