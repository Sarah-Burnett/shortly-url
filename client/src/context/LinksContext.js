import React, { useState, createContext } from 'react';

export const LinksContext = createContext();

export const LinksProvider = (props) => {
    const [ links, setLinks] = useState([]);

    return (
        <LinksContext.Provider value={[links, setLinks]}>
            {props.children}
        </LinksContext.Provider>
    )
};