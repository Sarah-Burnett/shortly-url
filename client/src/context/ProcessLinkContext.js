import React, { useState, createContext } from 'react';

export const ProcessLinkContext = createContext();

export const ProcessLinkProvider = (props) => {
    const [ processLink, setProcessLink] = useState({
        link: [],
        process: false
    });
    return (
        <ProcessLinkContext.Provider value={[processLink, setProcessLink]}>
            {props.children}
        </ProcessLinkContext.Provider>
    )
};