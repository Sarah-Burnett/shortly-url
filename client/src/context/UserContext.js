import React, { useState, createContext } from 'react';

export const UserContext = createContext();

export const UserProvider = (props) => {
    const [ user, setUser] = useState({
        isAuth: false,
        firstName: undefined,
        id: undefined,
        token: undefined,
    });
    
    return (
        <UserContext.Provider value={[user, setUser]}>
            {props.children}
        </UserContext.Provider>
    )
};