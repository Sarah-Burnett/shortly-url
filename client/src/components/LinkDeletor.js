import React, { useState, useContext } from 'react';
import { Button, PrimaryButton, LinkModal, Error } from '../Styles';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
import { LinksContext } from '../context/LinksContext';
import { LinkUpdateContext } from '../context/LinkUpdateContext';
import axios from 'axios';

export const LinkDeletor = () => {
    const [ links, setLinks ] = useContext(LinksContext);
    const [ linkUpdate, setLinkUpdate ] = useContext(LinkUpdateContext);
    const [ error, setError ] = useState();
    const { link } = linkUpdate;
    const deleteLink = (event) => {
        axios.delete(`/api/links/${link._id}`)
        .then( res => {setLinkUpdate(''); setLinks(prev => prev.filter( link => link !== linkUpdate.link ))})
        .catch( err => setError('Delete failed, please try again'));
    }
    return (
            <LinkModal>
                <h3>Delete Shortly Link</h3>
                <p>{link.full}</p>
                <p>{`https://shortly/${link.short}`}</p>
                { error && <Error><FontAwesomeIcon icon={faExclamationTriangle}/> {error}</Error> }
                <PrimaryButton id="update" type="button" onClick={() => deleteLink()}>Delete</PrimaryButton>
                <Button type="button" onClick={() => setLinkUpdate('')}>Cancel</Button>
            </LinkModal> 
    )
}
