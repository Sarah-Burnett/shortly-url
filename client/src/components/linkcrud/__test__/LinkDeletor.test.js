import React, { useState } from "react";
import {
	rendersWithoutCrashing,
	matchesSnapshot,
	ComponentWrapper,
} from "../../../setupTests";
import LinkDeletor from "../LinkDeletor";
import { UserContext } from "../../../context/UserContext";
import { LinksProvider } from "../../../context/LinksContext";
import { FlashProvider } from "../../../context/FlashContext";
import { ProcessLinkProvider } from "../../../context/ProcessLinkContext";

// const LinkDeletorWrapper = () => {
//     const [ user, setUser ] = useState({
//         isAuth: true,
//         id: "ha4skdfh45asdlkfh5aksf",
//         token: "Bearer adsfjla542fdjlasd.dfakdsfj24422lkasfj.htjk5445rhk5ajdsf",
//     });
//     return (
//         <UserContext.Provider value={[user, setUser]}>
//             <LinksProvider>
//                 <FlashProvider>
//                     <ProcessLinkProvider>
//                         <LinkDeletor/>
//                     </ProcessLinkProvider>
//                 </FlashProvider>
//             </LinksProvider>
//         </UserContext.Provider>
//     )
// };

const LinkDeletorWrapper = () => ComponentWrapper(LinkDeletor);

it("renders link deletor without crashing", () =>
	rendersWithoutCrashing(LinkDeletorWrapper));

it("link deletor matches snapshot", () => matchesSnapshot(LinkDeletorWrapper));

//TODO: check deleting link deletes link - integration
