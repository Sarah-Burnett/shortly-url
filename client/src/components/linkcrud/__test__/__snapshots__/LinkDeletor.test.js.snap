// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link deletor matches snapshot 1`] = `
<form
  className="sc-fzozJi blHhKY"
  onSubmit={[Function]}
>
  <h3>
    Delete Shortly Link
  </h3>
  <p />
  <p>
    https://short.ly/undefined
  </p>
  <button
    className="sc-AxirZ dVJbcR"
    id="delete"
  >
    Delete
  </button>
  <button
    className="sc-AxjAm kxpUMR"
    onClick={[Function]}
    type="button"
  >
    Close
  </button>
</form>
`;
